### DISCUSSION TITLE
Collection Literals: No target type?

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#7511] [Q&A] [answered]
Is this supposed to work when collection literals are officially released?
```
var Now = DateTimeOffset.UtcNow;
var Created_At = Members.Created_At_Computed() ?? Now;
var Updated_At = Members.Updated_At_Computed() ?? Now;

//This Line:
Created_At = [Created_At, Now].Min();
```

**Comment by HaloFour** [comment]
https://github.com/dotnet/csharplang/issues/5354#issuecomment-1697801770

> The plan is to aim for c#13 fit dictionary literals and natural types.

**Reply by TonyValenti** [reply]
Thanks. I was hopeful that extension methods would contribute to target type resolution.

**Reply by CyrusNajmabadi** [reply]
Not in 12

**Comment by jnm2** [comment, marked as answer]
LDM Jul 12 decided not to pursue extension methods specifically for collection expressions for now, but rather for a wider set of expressions including lambda expressions, to be explored for C# 13: https://github.com/dotnet/csharplang/blob/main/meetings/2023/LDM-2023-07-12.md#extension-methods

**Comment by davidkarlsson** [comment]
Hope this gets added in C# 13. I was suprised when I found out I couldn't chain my collection expressions with `Where` and the like.

**Reply by KennethHoff** [reply]
That is a separate issue, not directly related to target typing.

**Reply by jnm2** [reply]
While it is technically separate, I feel safe saying the two design conversations are related, one affecting the other thus far.
